### ISSUE TITLE
get counts of jobs on state

### ISSUE THREAD

**Issue by miguelEsteban** [imranmomin/Hangfire.AzureDocumentDB#58]
Hi @imranmomin ,
first of all, thanks for such a great job.
we have a very large database and we have found a problem in the method DocumentDbMonitoringApi.GetStatistics().
to retrieve the counters you have to read the entire database and the query takes about 4 min.

Changing to a query by state, the performance improves a lot, 1 sec.
so we have loaded the variable "result" reusing the method "GetNumberOfJobsByStateName"

results = keys.Select(x => new { Key = x.ToLower(), Value = GetNumberOfJobsByStateName(x) }).Where(x => x.Value > 0).ToDictionary(i => i.Key, i => i.Value);

![image](https://user-images.githubusercontent.com/9869085/96290566-19d23280-0fe7-11eb-80cb-e0bc373f6b1c.png)

**Comment by imranmomin** [comment]
@miguelEsteban - Yeah, this will reduce the time. 

I have also, another repo for .NET Core where I have used the **GroupBy** within the query. Have a look if that works or else you can raise the PR for the current changes

https://github.com/imranmomin/Hangfire.AzureCosmosDb/blob/1ed52d4ff05c34ba11654ca20c27786bd2c2ee13/src/CosmosDbMonitoringApi.cs#L123

**Comment by miguelEsteban** [comment]
Hi @imranmomin !
I have tested the cosmosDb version and it improves a lot compared to
documentDb. But even so, better times are obtained by making individual
querys by status.
in a little performance test I get these values.


   1. AzureDocumentDB
      1. original code: first access *28.992* ms. rest *28.500 *ms.
      2. Refactor: first access 532 ms. rest 230 ms
      2. AzureCosmosDb
      1. original code:  first access 4113 ms. rest 900 ms
      2. Refactor: first access 880 ms. rest 230 ms


as the method is called in each request, the sense of speed of the
dashboard changes a lot.

I have submitted the pull request to the documentDB repository. Same change
can be applied to cosmosDb repository.

Greetings,

Miguel Esteban.
